torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -3869,8 +3869,8 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
     return work;
   } else {
     at::cuda::CUDAGraph::dec_pending_event_queries();
-    return nullptr;
```

Comment:
@eqy regarding your question, I think my point was that becuase we do not enqueue the work object into the watchdog, then usercode calling `work.is_completed()` will always return false.  The watchdog (and workEnqueue) actually should not affect the behavior of work.wait() afaiu.

It's more common to use `work.wait()` than `work.is_completed()`, so making sure .wait works correctly is most critical. 

but i think if there is user-code control-flow based on work-completion during capture, this will be broken.

What if we modify 'work.is_completed()' api so that it throws an exception if CaptureStatus is active?

Reply:
I think that's a reasonable addition, @Aidyn-A, wdyt?

Reply:
> What if we modify 'work.is_completed()' api so that it throws an exception if CaptureStatus is active?

I second with Ke. It will make collective and p2p ops uncapturable because `wait()` calls `isCompleted()`. https://github.com/pytorch/pytorch/blob/0bb6408f47c92efc2d45bc89ebd01d8db24ca85c/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L746